pkg/stanza/fileconsumer/attrs/owner_other.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !windows
+
+package attrs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/attrs"
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"syscall"
+)
+
+func (r *Resolver) addOwnerInfo(file *os.File, attributes map[string]any) (err error) {
+	var fileInfo, errStat = file.Stat()
+	if errStat != nil {
+		return fmt.Errorf("resolve file stat: %w", err)
+	}
+	var fileStat = fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
I'm a bit concerned this could be fragile. I found some discussion about `Stat_t` not being present on some operating systems (most notably Windows, but not exclusively) but I can't find a definitive list of which OS we can count on. 

Can we capture the `ok` from this assertion and check it?